### PR TITLE
Add full escaping for shell commands usage

### PR DESCRIPTION
**Pull request by mjura**

### CODE REVIEW

File: chef/cookbooks/ceph/providers/pool.rb [hunk 1]
```diff
@@ -24,8 +24,8 @@ def get_pg_num(ceph_conf, admin_keyring)
 end
 
 def create_pool(ceph_conf, admin_keyring, pool_name, pg_num)
-  Chef::Log.info("Creating ceph pool '#{pool_name}' with pg number '#{pg_num}'")
-  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} osd pool create '#{pool_name}' '#{pg_num}'"
+  Chef::Log.info("Creating ceph pool #{pool_name} with pg number #{pg_num}")
+  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} osd pool create #{pool_name} #{pg_num}"
```
Comment (vuntz) [inline chef/cookbooks/ceph/providers/pool.rb]: @mjura as I said on IRC, if you remove the quotes, then you need full escaping. If you keep the quotes, then you need to escape quotes inside the variable

Comment (mjura) [inline chef/cookbooks/ceph/providers/pool.rb]: @vuntz I hope that you will be fine now :)

File: chef/cookbooks/ceph/providers/pool.rb [hunk 2]
```diff
@@ -24,8 +24,8 @@ def get_pg_num(ceph_conf, admin_keyring)
 end
 
 def create_pool(ceph_conf, admin_keyring, pool_name, pg_num)
-  Chef::Log.info("Creating ceph pool '#{pool_name}' with pg number '#{pg_num}'")
-  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} osd pool create '#{pool_name}' '#{pg_num}'"
+  Chef::Log.info("Creating ceph pool \"#{pool_name}\" with pg number #{pg_num}")
+  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} osd pool create \"#{pool_name}\" #{pg_num}"
```
Comment (vuntz) [inline chef/cookbooks/ceph/providers/pool.rb]: That's actually worse ;-) You still need to escape quotes (`"` now), but on top of that, you need  to escape `$` :-)

Comment (mjura) [inline chef/cookbooks/ceph/providers/pool.rb]: Once again updated :)

**Comment by rhafer** [comment]
Now the commit message doesn't fit the commit anymore :). But apart from that I think this one is fine.

File: chef/cookbooks/ceph/providers/pool.rb [hunk 3]
```diff
@@ -24,8 +24,8 @@ def get_pg_num(ceph_conf, admin_keyring)
 end
 
 def create_pool(ceph_conf, admin_keyring, pool_name, pg_num)
-  Chef::Log.info("Creating ceph pool '#{pool_name}' with pg number '#{pg_num}'")
-  cmd = "ceph -k #{admin_keyring} -c #{ceph_conf} osd pool create '#{pool_name}' '#{pg_num}'"
+  Chef::Log.info("Creating ceph pool #{pool_name} with pg number #{pg_num}")
+  cmd = ["ceph","-k",admin_keyring,"-c",ceph_conf,"osd","pool","create",pool_name,pg_num]
```
Comment (vuntz) [inline chef/cookbooks/ceph/providers/pool.rb]: Ah, nice. Just missing some spaces after each `,` :-)

**Comment by mjura** [comment]
Comments were fixed and PR was rebased

**Comment by tboerger** [comment]
LGTM

**Comment by rhafer** [comment]
+1

**Pull request merged**